beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -125,6 +128,20 @@ func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, ms
 		return pubsub.ValidationIgnore
 	}
 
+	// if the block slot does not belong to current slot, then queue the block for future processing.
+	currentSlot := s.cfg.Chain.CurrentSlot()
+	if blk.Block().Slot() > currentSlot {
```

Comment:
so this also queues blocks which are < 500 ms early. We should be processing them here instead of inserting them into the queue here.

Reply:
Good catch. Fixed this. But it is difficult to unit test this. It is hard to make the block arrive between `curren_time`  and `current_time + MAXIMUM_GOSSIP_CLOCK_DISPARITY `.